### ISSUE TITLE
Report "HTTP Proxy doesn't work for site requests action"

### ISSUE THREAD

**Issue by K1rakishou** [K1rakishou/Kuroba-Experimental#614]
UUID: 1417e0b723e0f858ced0e3c965ca1353
Build flavor: fdroid
Version name: v1.0.4
OS info: Android 10, sdk version: 29
Report description: I only tried this with 4chan enabled.
I can't tell if it works or not with the other actions but it definitely doesn't work with posting because I still get my normal flag when I post.

**Comment by K1rakishou** [comment]
Unfortunately I can't find working proxies to test it properly.

**Comment by Paskalo94** [comment]
> Unfortunately I can't find working proxies to test it properly.

 I can email you a working one if that's okay with you. Although, you will need to give me your IP address to temporarily allow usage of it.

**Comment by K1rakishou** [comment]
Just managed to find a working one and it seems to work. It was a SOCKS4/SOCKS5 proxy. Posted successfully on pol and the flag was different from mine - https://boards.4chan.org/pol/thread/344969912#p344997863

![image](https://user-images.githubusercontent.com/10332458/139058019-ef24273e-3d55-4b32-8e87-add82d274f42.png)


Did you restart the app after adding the proxy? Did you check the "Enable for site requests" checkbox? Did you select the correct site? And did you use the correct proxy type?

**Comment by Paskalo94** [comment]
Yes to all your questions. if I set my proxy to socks it doesn't even load threads because it's not a socks proxy.

Also I just realized the proxy works fine on /pol/ but not on /int/ or /sp/ could you test it yourself too? I guess it's something to do with them being on different domains

**Comment by K1rakishou** [comment]
Alright I think I found the reason why it doesn't work. Posting on 4channel uses sys.4channel.org endpoint and that endpoint wasn't checked in places where I'm trying to figure out the site type based on the url which made the replies not working with proxies on 4channel.org boards. I didn't test it (because couldn't find proxies) but it "should" work. If it doesn't (in the next update) then I will open this issue again.

Fixed in https://github.com/K1rakishou/Kuroba-Experimental/commit/8315bf08fa0f50b9bbbbc0445a25502d33cc8a46.
